### ISSUE TITLE
Not installing recursively

### ISSUE THREAD

**Issue by RobinMalfait** [maxleiko/npmi#13]
## SCENARIO

TL;DR NPM Doesn't install the dependencies recursively after second install in different folder

I have a plugins folder, and in the plugins folder I have a folder per plugin:

```
e.g.:
.
├── smithers-currency
│   ├── index.js
|   ├── node_modules
│   └── package.json
├── smithers-devtools
│   ├── index.js
|   ├── node_modules
│   └── package.json
└── smithers-font-awesome
    ├── index.js
    ├── node_modules
    └── package.json
```

This way, each plugin has its own dependencies without conflicts.
Now the problem is when I install something in smithers-currency (which depends on react) it works
When I install smithers-font-awesome after that, it doesn't work (also depends on react)

When I restart the application, and try to re-install the smithers-font-awesome plugin it works.

I tried setting some configs like force: true, to skip the cache, but I wonder what I'm doing wrong
## Something strange

I see this in my terminal; it looks like it didn't exit?

``` sh
⸨                 ░⸩ ⠙ runTopLevelLifecycles: sill install printInstalled
```
## Code I Used

``` js
import npmi from 'npmi'

const options = {
  name: moduleName,
  version: moduleVersion,
  path,
  forceInstall: true,
  npmLoad: {
    loglevel: 'warn',
    progress: true,
    force: true,
    json: true,
    production: true
  }
}

return new Promise((resolve, reject) => {
  npmi(options, (err, result) => {
    if (err) {
      reject(err)
    }

    resolve({ result, path })
  })
})
```
### Actual Output (example)

``` json
{
  "name": "plugin-smithers-font-awesome",
  "dependencies": {
    "smithers-font-awesome": {
      "version": "0.1.0",
      "from": "smithers-font-awesome@latest",
      "resolved": "https://registry.npmjs.org/smithers-font-awesome/-/smithers-font-awesome-0.1.0.tgz",
      "dependencies": {
        "react": {
          "version": "15.3.1",
          "from": "react@^15.0.0",
          "resolved": "https://registry.npmjs.org/react/-/react-15.3.1.tgz"
        }
      }
    }
  }
}
```
### Expected Output (example)

``` json
{
  "name": "plugin-smithers-currency",
  "dependencies": {
    "smithers-currency": {
      "version": "0.1.3",
      "from": "smithers-currency@latest",
      "resolved": "https://registry.npmjs.org/smithers-currency/-/smithers-currency-0.1.3.tgz",
      "dependencies": {
        "react": {
          "version": "15.3.1",
          "from": "react@>=15.0.0 <16.0.0",
          "resolved": "https://registry.npmjs.org/react/-/react-15.3.1.tgz",
          "dependencies": {
            "fbjs": {
              "version": "0.8.4",
              "from": "fbjs@>=0.8.4 <0.9.0",
              "resolved": "https://registry.npmjs.org/fbjs/-/fbjs-0.8.4.tgz",
              "dependencies": {
                "core-js": {
                  "version": "1.2.7",
                  "from": "core-js@>=1.0.0 <2.0.0",
                  "resolved": "https://registry.npmjs.org/core-js/-/core-js-1.2.7.tgz"
                },
                "immutable": {
                  "version": "3.8.1",
                  "from": "immutable@>=3.7.6 <4.0.0",
                  "resolved": "https://registry.npmjs.org/immutable/-/immutable-3.8.1.tgz"
                },
                "isomorphic-fetch": {
                  "version": "2.2.1",
                  "from": "isomorphic-fetch@>=2.1.1 <3.0.0",
                  "resolved": "https://registry.npmjs.org/isomorphic-fetch/-/isomorphic-fetch-2.2.1.tgz",
                  "dependencies": {
                    "node-fetch": {
                      "version": "1.6.0",
                      "from": "node-fetch@>=1.0.1 <2.0.0",
                      "resolved": "https://registry.npmjs.org/node-fetch/-/node-fetch-1.6.0.tgz",
                      "dependencies": {
                        "encoding": {
                          "version": "0.1.12",
                          "from": "encoding@>=0.1.11 <0.2.0",
                          "resolved": "https://registry.npmjs.org/encoding/-/encoding-0.1.12.tgz",
                          "dependencies": {
                            "iconv-lite": {
                              "version": "0.4.13",
                              "from": "iconv-lite@>=0.4.13 <0.5.0",
                              "resolved": "https://registry.npmjs.org/iconv-lite/-/iconv-lite-0.4.13.tgz"
                            }
                          }
                        },
                        "is-stream": {
                          "version": "1.1.0",
                          "from": "is-stream@>=1.0.1 <2.0.0",
                          "resolved": "https://registry.npmjs.org/is-stream/-/is-stream-1.1.0.tgz"
                        }
                      }
                    },
                    "whatwg-fetch": {
                      "version": "1.0.0",
                      "from": "whatwg-fetch@>=0.10.0",
                      "resolved": "https://registry.npmjs.org/whatwg-fetch/-/whatwg-fetch-1.0.0.tgz"
                    }
                  }
                },
                "promise": {
                  "version": "7.1.1",
                  "from": "promise@>=7.1.1 <8.0.0",
                  "resolved": "https://registry.npmjs.org/promise/-/promise-7.1.1.tgz",
                  "dependencies": {
                    "asap": {
                      "version": "2.0.4",
                      "from": "asap@>=2.0.3 <2.1.0",
                      "resolved": "https://registry.npmjs.org/asap/-/asap-2.0.4.tgz"
                    }
                  }
                },
                "ua-parser-js": {
                  "version": "0.7.10",
                  "from": "ua-parser-js@>=0.7.9 <0.8.0",
                  "resolved": "https://registry.npmjs.org/ua-parser-js/-/ua-parser-js-0.7.10.tgz"
                }
              }
            },
            "loose-envify": {
              "version": "1.2.0",
              "from": "loose-envify@>=1.1.0 <2.0.0",
              "resolved": "https://registry.npmjs.org/loose-envify/-/loose-envify-1.2.0.tgz",
              "dependencies": {
                "js-tokens": {
                  "version": "1.0.3",
                  "from": "js-tokens@>=1.0.1 <2.0.0",
                  "resolved": "https://registry.npmjs.org/js-tokens/-/js-tokens-1.0.3.tgz"
                }
              }
            },
            "object-assign": {
              "version": "4.1.0",
              "from": "object-assign@>=4.1.0 <5.0.0",
              "resolved": "https://registry.npmjs.org/object-assign/-/object-assign-4.1.0.tgz"
            }
          }
        }
      }
    }
  }
}
```

**Comment by maxleiko** [comment]
Resolving "conflicts" as you say is the job of `npm`. You should not be concerned by that. If you use `npmi@^3` then it will use your global `npm`. So if you have `npm@^3` installed, you will get a flattenned tree at the  root of the folder you aimed for using `npmi`.  

I might have misunderstood your issue there. Could you reformulate maybe?

**Comment by maxleiko** [comment]
@RobinMalfait ping

**Comment by RobinMalfait** [comment]
I will come back to this, currently studying for an exam

**Comment by maxleiko** [comment]
Long time no see.

**Comment by RobinMalfait** [comment]
Woops, I'm sorry for not closing or coming back to the issue I totally forgot about it. The project where I was using npmi is abandoned so I don't use it anymore.

Sorry about me being sloppy and not coming back, thanks for the great work of this package @maxleiko !
